tests/lax_numpy_test.py

```diff
@@ -2182,12 +2182,15 @@ def testAstype(self):
           jtu.format_shape_dtype_string(shape, a_dtype), dtype),
       "shape": shape, "a_dtype": a_dtype, "dtype": dtype}
       for shape in [(8,), (3, 8)]  # last dim = 8 to ensure shape compatibility
-      for a_dtype in [onp.bool_, onp.uint8, onp.float16, onp.int32, onp.float64]
-      for dtype in [onp.bool_, onp.int8, onp.int16, onp.float32, onp.float64]))
-  @jtu.skip_on_devices("tpu")
+      for a_dtype in (default_dtypes + unsigned_dtypes + bool_dtypes)
+      for dtype in (default_dtypes + unsigned_dtypes + bool_dtypes)))
   def testView(self, shape, a_dtype, dtype):
-    if not FLAGS.jax_enable_x64 and a_dtype == onp.float64 or dtype == onp.float64:
-      self.skipTest("x64 types are disabled by jax_enable_x64")
+    if jtu.device_under_test() == 'tpu':
+      if jnp.dtype(a_dtype).itemsize in [1, 2] or jnp.dtype(dtype).itemsize in [1, 2]:
+        self.skipTest("arr.view() not supported on TPU for 8- or 16-bit types.")
+    if not FLAGS.jax_enable_x64:
+      if jnp.dtype(a_dtype).itemsize == 8 or jnp.dtype(dtype).itemsize == 8:
+        self.skipTest("x64 types are disabled by jax_enable_x64")
     rng = jtu.rand_default(self.rng())
```

Comment:
I worry these (existing) tests might be a little skimpy. In particular `jtu.rand_default` is based on `np.random.RandomState.randn`, so the tests are only checking conversions for values near zero.

At the very least, I think it would be a good idea to generate integers from the full valid range. Generating a diverse set of random floats might be a bit tricker (I'm thinking of issues like multiple NaN values, whose semantics XLA may or may not exactly preserve)

Reply:
Good points - I changed the test to generate numbers using the full range of available bits for each type.

Perhaps it would be useful to add another test that round-trips for specific special-case inputs.

Reply:
Pathological floats are the only nasty cases that immediately come to mind, i.e., the duplicate representations for floating point zero, inf, -inf and nan. It seems _possible_ that int32 -> float32 -> int32 could fail to roundtrip if the float value is pathological enough.

Reply:
Added some tests of pathological floats.